components/brave_wallet/browser/solana_tx_manager_unittest.cc

```diff
@@ -322,4 +370,183 @@ TEST_F(SolanaTxManagerUnitTest, AddAndApproveTransaction) {
+  auto mojom_instruction = mojom::SolanaInstruction::New(
+      kSolanaSystemProgramId, std::move(account_metas), data);
+
+  std::vector<mojom::SolanaInstructionPtr> instructions;
+  instructions.push_back(std::move(mojom_instruction));
+  auto tx_data = mojom::SolanaTxData::New(
+      "", from_account, to_account, "", 10000000, 0,
+      mojom::TransactionType::SolanaSystemTransfer, std::move(instructions));
+
+  TestMakeSystemProgramTransferTxData(from_account, to_account, 10000000,
+                                      std::move(tx_data),
+                                      mojom::SolanaProviderError::kSuccess, "");
+
+  TestMakeSystemProgramTransferTxData(
+      "", to_account, 10000000, nullptr,
+      mojom::SolanaProviderError::kInternalError,
+      l10n_util::GetStringUTF8(IDS_WALLET_INTERNAL_ERROR));
+  TestMakeSystemProgramTransferTxData(
+      from_account, "", 10000000, nullptr,
+      mojom::SolanaProviderError::kInternalError,
+      l10n_util::GetStringUTF8(IDS_WALLET_INTERNAL_ERROR));
+}
+
+TEST_F(SolanaTxManagerUnitTest, MakeTokenProgramTransferTxData) {
+  // Test receiving associated token account exists.
+  std::string from_wallet_address =
+      "BrG44HdsEhzapvs8bEqzvkq4egwevS3fRE6ze2ENo6S8";
+  std::string to_wallet_address =
+      "JDqrvDz8d8tFCADashbUKQDKfJZFobNy13ugN65t1wvV";
+  std::string spl_token_mint_address =
+      "AQoKYV7tYpTrFZN6P5oUufbQKAUr9mNYGe1TTJC9wajM";
+
+  auto from_associated_token_account = SolanaKeyring::GetAssociatedTokenAccount(
+      spl_token_mint_address, from_wallet_address);
+  ASSERT_TRUE(from_associated_token_account);
+  auto to_associated_token_account = SolanaKeyring::GetAssociatedTokenAccount(
+      spl_token_mint_address, to_wallet_address);
+  ASSERT_TRUE(to_associated_token_account);
+
+  const std::vector<uint8_t> data = {3, 128, 150, 152, 0, 0, 0, 0, 0};
+
+  auto solana_account_meta1 = mojom::SolanaAccountMeta::New(
+      *from_associated_token_account, false, true);
+  auto solana_account_meta2 =
+      mojom::SolanaAccountMeta::New(*to_associated_token_account, false, true);
+  auto solana_account_meta3 =
+      mojom::SolanaAccountMeta::New(from_wallet_address, true, false);
+  std::vector<mojom::SolanaAccountMetaPtr> account_metas;
+  account_metas.push_back(std::move(solana_account_meta1));
+  account_metas.push_back(std::move(solana_account_meta2));
+  account_metas.push_back(std::move(solana_account_meta3));
+  auto mojom_transfer_instruction = mojom::SolanaInstruction::New(
+      kSolanaTokenProgramId, std::move(account_metas), data);
+
+  std::vector<mojom::SolanaInstructionPtr> instructions;
+  instructions.push_back(mojom_transfer_instruction.Clone());
+  auto tx_data = mojom::SolanaTxData::New(
+      "", from_wallet_address, to_wallet_address, spl_token_mint_address, 0,
+      10000000, mojom::TransactionType::SolanaSPLTokenTransfer,
+      std::move(instructions));
+
+  // Owner is not token program account.
```

Comment:
this comment is incorrect

Reply:
Fixed in https://github.com/brave/brave-core/pull/12729/commits/1ce4e199f80ca096af5836e593bb407bc1817b74, thanks